### PR TITLE
Add further missing C, C++ dependencies to packages

### PR DESCRIPTION
**Pull request by alalazo**
Extracted from #45189

### CODE REVIEW

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/googletest/package.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ class Googletest(CMakePackage):
     version("1.7.0", sha256="f73a6546fdf9fce9ff93a5015e0333a8af3062a152a9ad6bcb772c96687016cc")
     version("1.6.0", sha256="5fbc058e5b662b9c86d93ac76fefb58eec89cbf26144b49669a38ecb62758447")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/googletest/package.py:31]: @alalazo How were these determined? There's no C code in googletest...

Comment (alalazo) [reply var/spack/repos/builtin/packages/googletest/package.py:31]: @sethrj See https://github.com/spack/spack/pull/47662#issuecomment-2484192915

Comment (sethrj) [reply var/spack/repos/builtin/packages/googletest/package.py:31]: Hah, a common question then: guess CMake is sloppy about compiler checking. Thanks!

Comment (alalazo) [reply var/spack/repos/builtin/packages/googletest/package.py:31]: See also:

https://github.com/google/googletest/blob/d144031940543e15423a25ae5a8a74141044862f/CMakeLists.txt#L6C1-L6C33

From [CMake docs](https://cmake.org/cmake/help/latest/command/project.html):

> By default C and CXX are enabled if no language options are given. Specify language NONE, or use the LANGUAGES keyword and list no languages, to skip enabling any languages.

Comment (sethrj) [reply var/spack/repos/builtin/packages/googletest/package.py:31]: Maybe we need a spackbot that automatically opens pull requests for sloppy packages 😂

**Pull request merged**